pyresample/boundary/boundary_sides.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2023 Pyresample developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Define the BoundarySides class."""
+
+import logging
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+class BoundarySides:
+    """A class to represent the sides of an area boundary.
+
+    The sides are stored as a tuple of 4 numpy arrays, each representing the
+    coordinate (geographic or projected) of the vertices of the boundary side.
+    The sides must be stored in the order (top, right, left, bottom),
+    which refers to the side position with respect to the coordinate array.
+    The first row of the coordinate array correspond to the top side, the last row to the bottom side,
+    the first column to the left side and the last column to the right side.
+    Please note that the last vertex of each side must be equal to the first vertex of the next side.
+    """
+    __slots__ = ['_sides']
```

Comment:
I mentioned this in the large PR and I'm curious what @mraspaud thinks, but I think this entire class can be replaced with a `namedtuple`. A namedtuple would provide all the same functionality *except* the `.vertices` property but I think you can subclass the namedtuple to add that if you need. You could also then do your own docstrings as you have them.

Reply:
A dataclass would be an alternative

Reply:
My understanding is that profiling done by some Python core devs have proven that data classes are not faster than any of the alternatives and in some cases are slower. I don't remember those cases and differences.